### PR TITLE
MYT-12 Save todo list data to local storage 

### PR DESCRIPTION
**Pull request by nottombraider**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { createTaskListItemMarkup } from './handlers';
 class App {
   constructor() {
     this.container = document.getElementById('app');
-    this.taskList = [];
+
```
Comment (yutro) [inline src/index.js]: new line is redundant

File: src/index.js [hunk 2]
```diff
@@ -51,16 +51,37 @@ class App {
     this.container.appendChild(addTask);
     this.container.appendChild(this.taskListContainer);
     this.container.appendChild(buttonDeleteCompletedTasks);
+
+    if (localStorage.toDoTaskList.length) {
```
Comment (yutro) [inline src/index.js]: const items =localStorage.getItem('toDoTaskList');
if(items && items.length) {
...
}

File: src/index.js [hunk 3]
```diff
@@ -51,16 +51,37 @@ class App {
...
+      this.taskList = this.getTaskListFromLocalStorage();
```
Comment (yutro) [inline src/index.js]: line 56 can be moved out of if statement to line 54 
checking if task list in localStorage should be done in getTaskListFromLocalStorage 
if ls empty, [] can be returned

File: src/index.js [hunk 4]
```diff
@@ -51,16 +51,37 @@ class App {
...
+      this.render();
+    } else {
+      this.taskList = [];
+    }
   }
 
+  saveTaskListToLocalStorage() {
+    if (this.taskList.length) {
```
Comment (yutro) [inline src/index.js]: redundant

File: src/index.js [hunk 5]
```diff
@@ -51,16 +51,37 @@ class App {
...
+  getTaskListFromLocalStorage() {
+    const dataFromLocalStorage = localStorage.getItem('toDoTaskList');
+    this.taskList = JSON.parse(dataFromLocalStorage);
+    return this.taskList;
+  }
 
   addToDo(task) {
     this.taskList.push(task);
+    this.saveTaskListToLocalStorage();
     this.render();
```
Comment (yutro) [inline src/index.js]: saving can be done after render

File: src/index.js [hunk 6]
```diff
@@ -51,16 +51,37 @@ class App {
...
   }
 
   removeToDos() {
     this.taskList = this.taskList.filter((item) => !item.done);
+    this.saveTaskListToLocalStorage();
     this.render();
```
Comment (yutro) [inline src/index.js]: saving can be done after render

File: src/index.js [hunk 7]
```diff
@@ -51,16 +51,37 @@ class App {
...
 
+  saveTaskListToLocalStorage() {
+    if (this.taskList.length) {
+      const taskListStr = JSON.stringify(this.taskList);
+      localStorage.setItem('toDoTaskList', taskListStr);
+    }
+  }
+
+  getTaskListFromLocalStorage() {
+    const dataFromLocalStorage = localStorage.getItem('toDoTaskList');
```
Comment (yutro) [inline src/index.js]: a key can be moved to ex. this.appKey in constructor

**Review by yutro — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 4 above)
Comment (nottombraider) [reply src/index.js]: done

File: src/index.js (hunk 5 above)
Comment (nottombraider) [reply src/index.js]: done

File: src/index.js (hunk 6 above)
Comment (nottombraider) [reply src/index.js]: done

File: src/index.js (hunk 7 above)
Comment (nottombraider) [reply src/index.js]: done

File: src/index.js (hunk 3 above)
Comment (nottombraider) [reply src/index.js]: done

File: src/index.js (hunk 2 above)
Comment (nottombraider) [reply src/index.js]: done

File: src/index.js (hunk 1 above)
Comment (nottombraider) [reply src/index.js]: done

File: src/index.js [hunk 8]
```diff
@@ -51,17 +51,35 @@ class App {
...
   }
 
+  getTaskListFromLocalStorage() {
+    const dataFromLocalStorage = localStorage.getItem(this.localStorageKey);
+    if (dataFromLocalStorage && dataFromLocalStorage.length) {
+      this.taskList = JSON.parse(dataFromLocalStorage);
+      return this.taskList;
+    }
+    return [];
+  }
```
Comment (yutro) [inline src/index.js]: use this.taskList.push(data from local storage)

File: src/index.js [hunk 9]
```diff
@@ -3,7 +3,7 @@ import { createTaskListItemMarkup } from './handlers';
 class App {
   constructor() {
     this.container = document.getElementById('app');
-    this.taskList = [];
```
Comment (yutro) [inline src/index.js:6]: needed

File: src/index.js [hunk 10]
```diff
@@ -51,17 +51,35 @@ class App {
     this.container.appendChild(addTask);
     this.container.appendChild(this.taskListContainer);
     this.container.appendChild(buttonDeleteCompletedTasks);
+
+    this.taskList = this.getTaskListFromLocalStorage();
```
Comment (yutro) [inline src/index.js]: assignment not needed this.getTaskListFromLocalStorage() would be enough

File: src/index.js (hunk 8 above)
Comment (nottombraider) [reply src/index.js]: done

File: src/index.js (hunk 9 above)
Comment (nottombraider) [reply src/index.js:6]: added

File: src/index.js (hunk 10 above)
Comment (nottombraider) [reply src/index.js]: done

File: src/index.js (hunk 8 above)
Comment (nottombraider) [reply src/index.js]: not needed

**Review by yutro — APPROVED** [review]

**Pull request merged**